docs/sample-v2-addon/rollup.config.mjs

```diff
@@ -0,0 +1,60 @@
+import { Addon } from '@embroider/addon-dev/rollup';
+import postcss from 'rollup-plugin-postcss';
+import typescript from 'rollup-plugin-ts';
+
+const addon = new Addon({
+  srcDir: 'src',
+  destDir: 'dist',
+});
+
+export default {
+  // This provides defaults that work well alongside `publicEntrypoints` below.
+  // You can augment this if you need to.
+  output: addon.output(),
+
+  plugins: [
+    // These are the modules that users should be able to import from your
+    // addon. Anything not listed here may get optimized away.
+    addon.publicEntrypoints([
+      'components/**/!(*.css.d).js',
+      'index.js',
+      'template-registry.js',
+    ]),
+
+    // These are the modules that should get reexported into the traditional
+    // "app" tree. Things in here should also be in publicEntrypoints above, but
+    // not everything in publicEntrypoints necessarily needs to go here.
+    addon.appReexports(['components/**/!(*.css.d).js']),
+
+    // Follow the V2 Addon rules about dependencies. Your code can import from
+    // `dependencies` and `peerDependencies` as well as standard Ember-provided
+    // package names.
+    addon.dependencies(),
+
+    // Implement CSS modules
+    postcss({
+      autoModules: false,
+      modules: {
+        generateScopedName: 'sample-v2-addon__[sha512:hash:base64:5]',
+        // generateScopedName: 'sample-v2-addon__[path][name]__[local]',
+      },
+    }),
```

Comment:
I found two ways to configure `rollup-plugin-postcss`. One, in which the addon maintainer generates a specific hash (perhaps they want a namespace based on the addon's name), is shown here.

The other, a simpler way, is:

```js
postcss({
  modules: true,
});
```

It's important to note that the order of the plugins matters. In TypeScript projects, `postcss()` must appear before `typescript()`. Otherwise, the addon's `build` script will result in an error:

```sh
[sample-v2-addon] rollup v3.21.0
[sample-v2-addon] bundles  → dist...
[sample-v2-addon] [!] (plugin Typescript) SyntaxError: /packages/sample-v2-addon/src/components/hello-world.css: Unexpected token (1:0)
[sample-v2-addon] 
[sample-v2-addon] > 1 | .container {
[sample-v2-addon]     | ^
[sample-v2-addon]   2 |   border: 1px solid steelblue;
[sample-v2-addon]   3 |   margin-bottom: 1rem;
[sample-v2-addon]   4 | }
[sample-v2-addon] src/components/hello-world.css (1:0)
```